tests/backup/backup_restore_basic_test.go

```diff
@@ -3189,3 +3190,145 @@ var _ = Describe("{DeleteNSDeleteClusterRestore}", func() {
+		})
+
+		Step("Registering cluster for backup", func() {
+			log.InfoD("Registering cluster for backup")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateApplicationClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			log.InfoD("Verifying cluster status for both source and destination clusters")
+			clusterStatus, err := Inst().Backup.GetClusterStatus(orgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, orgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+			clusterStatus, err = Inst().Backup.GetClusterStatus(orgID, destinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", destinationClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", destinationClusterName))
+		})
+
+		Step("Taking alternate backups of application from source cluster to both S3 and NFS backup locations", func() {
+			log.InfoD("Taking alternate backups of application from source cluster to both S3 and NFS backup locations")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, bkpNamespaces)
+			for i := 0; i < numberOfAlternateBackups; i++ {
+				for locationUID, locationName := range backupLocationMap {
+					log.InfoD("Creating backup using the backup location of [%s]", locationName)
+					backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, RandomString(10))
+					backupNames = append(backupNames, backupName)
+					err = CreateBackupWithValidation(ctx, backupName, SourceClusterName, locationName, locationUID, appContextsToBackup, labelSelectors, orgID, sourceClusterUid, "", "", "", "")
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", backupName))
+					log.InfoD("Verifying the type of backup")
+					//First backup for each backup location must be a full backup, rest should be incremental
+					if i == 0 {
+						err = IsFullBackup(backupName, orgID, ctx)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying if backup [%s] is a full backup", backupName))
+					} else {
+						err = IsFullBackup(backupName, orgID, ctx)
+						if strings.Contains(err.Error(), "is an incremental backup") {
+							log.InfoD("The backup [%s] is an incremental backup", backupName)
+						} else {
+							log.FailOnError(err, "Issue with created backup.")
+						}
+					}
+				}
+			}
+			log.Infof("List of backups - %v", backupNames)
+		})
```

Comment:
It appears that the test case is taking 26 minutes, which I believe is a bit too long. How about we parallelize this step?